src/js/shopify.js

```diff
@@ -0,0 +1,44 @@
+import Client from 'shopify-buy';
+
+// Initialize a client:
+const client = Client.buildClient({
+     domain: import.meta.env.VITE_SHOPIFY_HOSTNAME,
+     storefrontAccessToken: import.meta.env.VITE_SHOPIFY_STOREFRONT_API_TOKEN,
+});
+
+// Create an empty checkout (@todo look at checking for existing checkout?)
+client.checkout.create().then((checkout) => {
+
+     const $buyButtons = document.querySelectorAll(".buy-button");
+
+     $buyButtons.forEach(button => {
+          button.addEventListener("click", () => {
+
+               updateButtonUIToLoading(button);
+
+               const lineItemsToAdd = [
+                    {
+                         variantId: 'gid://shopify/ProductVariant/' + button.dataset.defaultVariantId,
+                         quantity: 1,
+                    }
+               ];
+
+               // Add an item to the checkout
+               client.checkout.addLineItems(checkout.id, lineItemsToAdd).then((checkout) => {
+                    // Do something with the updated checkout
+                    updateButtonUIToSuccess(button);
+                    console.log(checkout.lineItems); // Array with one additional line item
+               });
+
+          });
+     });
+
+});
+
+function updateButtonUIToLoading(button) {
+     button.disabled = true;
+}
+
+function updateButtonUIToSuccess(button) {
+     button.disabled = false;
+}
```

Comment:
Heads up that the button can now be stuck in these states if the call to add the line item to the Shopify cart fails.